centml/cli/cluster.py

```diff
@@ -158,88 +187,45 @@ def get(type, id):
         )
 
 
-# Define common deployment
-def common_options(func):
-    hw_to_id_map, _ = get_hw_to_id_map()
-    func = click.option("--name", "-n", prompt="Name", help="Name of the deployment")(func)
-    func = click.option("--image", "-i", prompt="Image", help="Container image")(func)
-    func = click.option(
-        "--hardware",
-        "-h",
-        prompt="Hardware",
-        type=click.Choice(list(hw_to_id_map.keys())),
-        help="Hardware instance type",
-    )(func)
-    return func
-
-
-# Define inference specific options
-def inference_options(func):
-    func = click.option("--port", "-p", prompt="Port", type=int, help="Port to expose")(func)
-    func = click.option(
-        "--env", type=InferenceEnvType(), help="Environment variables in the format KEY=VALUE", multiple=True
-    )(func)
-    func = click.option("--min_replicas", default="1", prompt="Min replicas", type=click.IntRange(1, 10))(func)
-    func = click.option("--max_replicas", default="1", prompt="Max replicas", type=click.IntRange(1, 10))(func)
-    func = click.option("--health", default="/", prompt="Health check", help="Health check endpoint")(func)
-    func = click.option("--is_private", default=False, type=bool, prompt="Is private?", help="Is private endpoint?")(
-        func
-    )
-    func = click.option("--timeout", prompt="Max concurrency", default=0, type=int)(func)
-    func = click.option("--command", type=str, required=False, default=None, help="Define a command for a container")(
-        func
-    )
-    func = click.option("--command_args", multiple=True, type=str, default=None, help="List of command arguments")(func)
-    return func
-
+@click.group(help="Create a new deployment")
+def create():
+    pass
 
-# Define compute specific options
-def compute_options(func):
-    func = click.option("--username", prompt="Username", type=str, help="Username")(func)
-    func = click.option("--password", prompt="Password", hide_input=True, type=str, help="password")(func)
-    func = click.option(
-        "--ssh_key", prompt="Add ssh key", default="", type=str, help="Would you like to add an SSH key?"
-    )(func)
-    return func
 
+# Custom click type that uses the singleton instance for validation
+class HardwareType(click.ParamType):
+    name = "hardware"
 
-# Main command group
-@click.group(help="Create a new deployment")
-@click.pass_context
-def create(ctx):
-    pass
+    def convert(self, value, param, ctx):
```

Comment:
**Note**: we have defined this as a common class so that it can be used as a type for inference and compute both commands to validate the choices, and during creation of request payload we can directly access the singleton instance because it will hold the response for the choices

Reply:
we can not use the same class HardwarePricingTier to validate the same here because click will initialize the parent class everytime because during code compilation it executes all the classes defined in click options and then we will not be able to avoid calling api multiple times, so thats why i have created it as a separate class just to check the validation part of the choices which will be eventually calling the validate method of HardwarePricingTier class